src/string/predicate.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt::Debug;
+
+use itertools::Itertools;
+
+use crate::{ArityPredicate, Predicate};
+
+use super::StringPosition;
+
+/// A predicate for matching a string.
+///
+/// We support two types of predicates: matching a character against a constant
+/// `char`, and comparing to character variables.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum StringPredicate {
+    /// Comparison between two bound values
+    BindingEq,
+    /// Comparison of a bound value with a constant char
+    ConstVal(char),
+}
+
+impl ArityPredicate for StringPredicate {
```

Comment:
I split off `ArityPredicate` at some point to simplify trait bounds (often we need to have an arity, but don't care about the predicate further than that).

This might change in the future